# HG changeset patch
# User kai@linux-pc
# Date 1205289182 14400
#      Tue Mar 11 22:33:02 2008 -0400
# Branch release-3-0-x
# Node ID 7817fd1f2237436c054d10cb1e279e79abd8d6d5
# Parent  7956ecd5fa4385fe3cccd0a29f170e8c7df8f0b3
Improvements for surface plots

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -2,16 +2,20 @@ 2008-03-11  John W. Eaton  <jwe@octave.o
 
 	* plot/ChangeLog: Handle axes linewidth property.
 
 	* plot/__go_draw_axes__.m (get_old_gnuplot_color): New subfunction.
 	Use it to replace repeated if/elseif blocks.
 
 2008-03-11  Kai Habel  <kai.habel@gmx.de>
 
+        * plot/__go_draw_axes__.m: Plot surfaces in front of axes.
+	Allow plotting of uniform colored mesh plots with and
+	w/o hidden line removal.
+
 	* plot/__go_draw_axes__.m: If facecolor property of a surface is
 	"none", don't use pm3d mode and set linestyle correctly. 
 
 2008-03-07  John W. Eaton  <jwe@octave.org>
 
 	* plot/contourf.m: Set axes layer property to "top".
 
 2008-03-06  John W. Eaton  <jwe@octave.org>
diff --git a/scripts/plot/__go_draw_axes__.m b/scripts/plot/__go_draw_axes__.m
--- a/scripts/plot/__go_draw_axes__.m
+++ b/scripts/plot/__go_draw_axes__.m
@@ -862,52 +862,54 @@ function __go_draw_axes__ (h, plot_strea
 	        zz(:,kk+1) = ydat(:,k);
 	        zz(:,kk+2) = zdat(:,k);
 	        zz(:,kk+3) = cdat(:,k);
 	        k++;
 	      endfor
 	      data{data_idx} = zz.';
 	    endif
 	    usingclause{data_idx} = "using ($1):($2):($3):($4)";
-	    ## fputs (plot_stream, "unset parametric;\n");
+            ## fputs (plot_stream, "unset parametric;\n");
 
 	    ## Interpolation does not work for flat surfaces (e.g. pcolor)
             ## and color mapping --> currently set empty.
             interp_str = "";
             flat_interp_face = (strncmp (obj.facecolor, "flat", 4)
 				|| strncmp (obj.facecolor, "interp", 6));
             flat_interp_edge = (strncmp (obj.edgecolor, "flat", 4)
 				|| strncmp (obj.edgecolor, "interp", 6));
 
 	    facecolor_none_or_white = (strncmp (obj.facecolor, "none", 4)
 				       || (isnumeric (obj.facecolor)
 					   && all (obj.facecolor == 1)));
-	    if (strncmp (obj.facecolor, "none", 4))
-	      hidden_removal = false;
-	    else
-	      if (isnan (hidden_removal))
-		hidden_removal = true;
-	      endif
-	    endif
+	    hidden_removal = false;
 	
             if (flat_interp_edge && facecolor_none_or_white)
 	      withclause{data_idx} = "with line palette";
 	      fputs (plot_stream, "unset pm3d\n");
+	      if (all (obj.facecolor == 1))
+                  hidden_removal = true;
+              endif
 	    elseif (facecolor_none_or_white)
 	      edgecol = obj.edgecolor;
 	      if (have_newer_gnuplot)
 		if (mono)
 		  colorspec = "";
 		else
 		  colorspec = sprintf ("linecolor rgb \"#%02x%02x%02x\"",
 				 round (255*edgecol));
 		endif
+		if (all (obj.facecolor == 1))
+                  hidden_removal = true;
+                endif
+		fputs(plot_stream,"unset pm3d;\n");
                 fprintf (plot_stream,
                          "set style line %d %s lw %f;\n",
                          data_idx, colorspec, obj.linewidth);
+		fputs(plot_stream,"set style increment user;\n");
               else
 		if (isequal (edgecol, [0,0,0]))
 		  typ = -1;
 		elseif (isequal (edgecol, [1,0,0]))
 		  typ = 1;
 		elseif (isequal (edgecol, [0,1,0]))
 		  typ = 2;
 		elseif (isequal (edgecol, [0,0,1]))
@@ -923,17 +925,17 @@ function __go_draw_axes__ (h, plot_strea
 		else
 		  typ = -1;
 		endif
                 fprintf (plot_stream,
                          "set style line %d lt %d lw %f;\n",
                          data_idx, typ, obj.linewidth);
 	      endif
 	      withclause{data_idx} = sprintf("with line linestyle %d", data_idx);
-	      fputs (plot_stream, "unset pm3d\n")
+	      fputs (plot_stream, "unset pm3d\n");
             endif
 
 	    if (have_newer_gnuplot)
 	      dord = "depthorder";
 	    else
 	      dord = "";
 	    endif
 
